### DISCUSSION TITLE
macOS: Wire makes a click noise and the App-Icon jumps even if "Do not disturb" is enabled

### DISCUSSION THREAD

**Discussion by KizzyCode** [wireapp/wire-desktop#5070] [Ideas]
## Environment info:
**Wire version:** 3.5.2910
**Wire for web version:** 2019.01.08.1320
**Operating system:** macOS Mojave 10.14.2 (18C54)
**Which antivirus software do you have installed:** None

## What steps will reproduce the problem?
 1. Activate "Do not disturb" in the notification center
 2. Close Wire's window and focus another app
 3. Receive a message

## What is the expected result?
Nothing (you shouldn't notice that you received a message)

## What is the actual result?
The app-icon in the dock jumps multiple times and Wire plays the "New message" click-sound.

## Please provide any additional information below:
This probably violates the App Store app guidelines so this bug *might* be more serious than it sounds.

**Comment by ffflorian** [comment]
Hi @KizzyCode, sorry for the late answer - honoring the DND mode on macOS is not possible out-of-the-box in Electron. They [recommend](https://github.com/electron/electron/blob/v4.2.12/docs/tutorial/notifications.md#do-not-disturb--session-state) using a 2-year-old library, which doesn't look very promising.
Maybe @pwnsdx can help out here?

Update: I found out that we could use [`macos-notification-state`](https://github.com/felixrieseberg/macos-notification-state) but it requires building in a native module.
